### DISCUSSION TITLE
Return a custom property to getStaticProps with getStaticPaths

### DISCUSSION THREAD

**Discussion by gbreux** [vercel/next.js#11738] [Help]
Hi there!

Is there a way to return a custom property to `getStaticProps` with `getStaticPaths`?

For example, I have this url: `/blog/[slug]` but the slug can't be used to fetch the article, instead I need to use an id (but the id is ugly and can't be in the url)

So I would need to return: 
```
export async function getStaticPaths() {
    return {
	params: {
		slug,
		id,
	},
    };
}
```

So I can fetch the article inside `getStaticProps`?

Thanks a lot

**Comment by janus-reith** [comment]
You could in theory fetch pairs of slugs and ids in getStaticPaths, but that connection would get lost, as your route just defines one parameter, which is the slug.

Simply pass the slug from getStaticPaths to getStaticProps and fetch the id or article there.
If you can't fetch by slug, fetch your list of id+slug pairs there and find the id by using the slug you passed from getStaticPaths.

**Reply by gbreux** [reply]
I guess that's a solution but it looks a little bit dirty to me :D 

Is there any reason why we can't pass any custom props that are not used in the dynamic routes?

**Comment by willemliu** [comment]
If the slug does not connect to the id then what would happen if a user lands on a URL? How can your code know what page to show the user?

**Reply by gbreux** [reply]
I know it at rendered time not at runtime so the user will land on a known and rendered url already I dont need to fetch the post at runtime

**Comment by charlesRepo** [comment]
Any updates on this issue?
It seems that it is not possible to return this in `getStaticPaths`
`posts/[slug].js`
```
return {
 params:{
  id: post.id,
  slug: post.slug
 }
}
```
and then use only the id in `getStaticProps` in the api
`wp-json/wp/v2/posts/${params.id}`

What alternatively can be done to achieve this, if only we can get posts from the API using ID and we need the slug for dynamic routing?

**Comment by nandorojo** [comment]
In my case, I had to restructure my backend to let me query by slug, rather than ID.
